### PR TITLE
Update Github workflow with JDK21

### PR DESCRIPTION
**Pull request by thachlp**
**Motivation**
Since Java 21 is an LTS version released on September 19, 2023. We should update the CI Github build.

Result
Close https://github.com/line/centraldogma/issues/911

### CODE REVIEW

File: .github/workflows/actions_build.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-12, windows-latest]
-        java: [17]
+        java: [17, 21]
```
Comment (ikhoon) [inline .github/workflows/actions_build.yml:31]: Should we upgrade `BUILD_JDK_VERSION` to 21 as well?

File: .github/workflows/actions_build.yml [hunk 2]
```diff
@@ -42,20 +42,22 @@ jobs:
             min-java: 11
           - java: 19
             os: ubuntu-latest
+          - java: 21
+            os: ubuntu-latest
```
Comment (ikhoon) [inline .github/workflows/actions_build.yml:46]: We usually use non-LTS for the latest version. Let's remove Java 19 version.
```suggestion
          - java: 21
            os: ubuntu-latest
```

Comment (thachlp) [reply .github/workflows/actions_build.yml]: Updated

File: .github/workflows/actions_build.yml (hunk 1 above)
Comment (thachlp) [reply .github/workflows/actions_build.yml:31]: Yes, I updated

Comment (jrhee17) [inline .github/workflows/actions_build.yml:31]: I prefer only 21 is run on mac/windows. The build is flaky enough as it is already 😅 

```suggestion
        java: [21]
```

**Review by jrhee17 — COMMENTED** [review]
Can we also modify to release from 21?

https://github.com/line/centraldogma/blob/2994712c79f041941d5fb5160368823b59832e29/build.gradle#L141

**Comment by minwoox** [comment]
Please address this comment. 🙇 
https://github.com/line/centraldogma/pull/927#pullrequestreview-1942025829
Does anyone know why we didn't use JDK 19 before?

**Comment by ikhoon** [comment]
Maybe it wasn't a LTS version or we are lazy? 😅

**Comment by thachlp** [comment]
> Can we also modify to release from 21?
> 
> https://github.com/line/centraldogma/blob/2994712c79f041941d5fb5160368823b59832e29/build.gradle#L141

Hi @jrhee17, updated 🙇

**Comment by thachlp** [comment]
> Please address this comment. 🙇 [#927 (review)](https://github.com/line/centraldogma/pull/927#pullrequestreview-1942025829) Does anyone know why we didn't use JDK 19 before?

Thanks, @minwoox, I missed that comment.

**Review by minwoox — APPROVED** [review]
Thanks a lot, @thachlp 😆

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

**Pull request merged**